firewood/src/storage/linear/historical.rs

```diff
@@ -4,28 +4,31 @@
 use std::collections::BTreeMap;
 use std::sync::Arc;
 
-use super::{layered::LayeredReader, LinearStore, ReadLinearStore};
+use super::{
+    layered::{Layer, LayeredReader},
+    ImmutableLinearStore,
+};
 
 /// A linear store used for historical revisions
 ///
 /// A [Historical] [LinearStore] supports read operations only
 #[derive(Debug)]
-pub(crate) struct Historical<P: ReadLinearStore> {
+pub(crate) struct Historical {
     /// (offset, value) for every area of this LinearStore that is modified in
     /// the revision after this one (i.e. `parent`).
     /// For example, if the first 3 bytes of this revision are [0,1,2] and the
     /// first 3 bytes of the next revision are [4,5,6] then this map would
     /// contain [(0, [0,1,2])].
-    pub(crate) changed_in_parent: BTreeMap<u64, Box<[u8]>>,
+    changed_in_parent: BTreeMap<u64, Box<[u8]>>,
```

Comment:
If there's a time to use `bytes::Bytes`, it's right now. It's the most used sub-crate by any "networking" crate (in other words, any crate that's copying around byte slices). It's in the top 50 most downloaded crates of all time for good reason. 

@danlaine, this comment isn't just for you.

Reply:
I'm not opposed but think that should be a separate PR

Reply:
We can create an issue for this and do it separately. The nice bit about `Bytes` is that you get nearly free copies and moves. We need this at commit time, when `old` becomes `was` in the newly created `LinearStore`.